aptos-move/aptos-vm/src/transaction_validation.rs

```diff
@@ -350,21 +356,38 @@ pub(crate) fn run_multisig_prologue(
     session: &mut SessionExt<impl AptosMoveResolver>,
     module_storage: &impl ModuleStorage,
     txn_data: &TransactionMetadata,
-    payload: &Multisig,
+    executable: TransactionExecutableRef,
+    multisig_address: AccountAddress,
     features: &Features,
     log_context: &AdapterLogSchema,
     traversal_context: &mut TraversalContext,
 ) -> Result<(), VMStatus> {
     let unreachable_error = VMStatus::error(StatusCode::UNREACHABLE, None);
-    let provided_payload = if let Some(payload) = &payload.transaction_payload {
-        bcs::to_bytes(&payload).map_err(|_| unreachable_error.clone())?
-    } else {
-        // Default to empty bytes if payload is not provided.
-        if features.is_abort_if_multisig_payload_mismatch_enabled() {
-            vec![]
-        } else {
-            bcs::to_bytes::<Vec<u8>>(&vec![]).map_err(|_| unreachable_error)?
-        }
+    // Question: Earlier the `provided_payload` was being calculated as bcs::to_bytes(MultisigTransactionPayload::EntryFunction(entry_function)).
+    // So, converting the executable to this format. Is this correct?
+    let provided_payload = match executable {
+        TransactionExecutableRef::EntryFunction(entry_function) => {
+            // TODO[Orderless]: See how to avoid cloning the entry function here.
+            bcs::to_bytes(&MultisigTransactionPayload::EntryFunction(
+                entry_function.clone(),
+            ))
+            .map_err(|_| unreachable_error.clone())?
+        },
+        TransactionExecutableRef::Empty => {
+            if features.is_abort_if_multisig_payload_mismatch_enabled() {
+                vec![]
+            } else {
+                bcs::to_bytes::<Vec<u8>>(&vec![]).map_err(|_| unreachable_error.clone())?
+            }
+        },
+        TransactionExecutableRef::Script(_) => {
+            // Question: Earlier, multisig transactions only supported EntryFunction payloads.
+            // So, what should we do if the executable for a multisig transaction is a script?
+            return Err(VMStatus::error(
+                StatusCode::FEATURE_NOT_ENABLED,
```

Comment:
this doesn't look like a good status code

Reply:
> was this abort or discard before?
Earlier, the multisig payload structure only had EntryFunctionPayload variant. There was not even a way to craft a ScriptPayload.

Reply:
Based on the above comments, should we use `FEATURE_UNDER_GATING` instead and discard the transactions?